### PR TITLE
Render QR Code

### PR DESCRIPTION
**Pull request by aschmois**
If the client mod is present, the qr code is generated in game, if not,
old behavior activates. Also added a config option for IP address or
domain name, so that we send something meaningful to remote clients.

### CODE REVIEW

**Comment by liach** [comment]
Yep but how it works servers that google was blocked? You see some countries cannot use google.

**Comment by aschmois** [comment]
Well, to be honest I didn't touch that, but the QR code is something to facilitate things, if there are unforeseen situations where the qr code can't be generated, typing in the values manually is still an option.

File: src/main/java/com/forgeessentials/core/ForgeEssentials.java [hunk 1]
```diff
@@ -287,5 +291,15 @@ public void setDebugMode(boolean debugMode)
     {
         this.debugMode = debugMode;
     }
+
```
Comment (luacs1998) [inline src/main/java/com/forgeessentials/core/ForgeEssentials.java]: No. I should have code already in place to check if a client has the FE client - if not I'll get it in place soon.

**Comment by aschmois** [comment]
Didn't see one, added one. I don't see what's wrong with this implementation either, I use the exact implementation for the client to check if the server is installed.

**Comment by luacs1998** [comment]
The server does not know whose modset it is. Your clientHasFE thing applies to the whole server, because you failed to differentiate between players.

Looking in PlayerInfo I guess I probably only implemented it halfway so I'll finish up.

**Comment by KaneHart** [comment]
Did you make sure there was no hanging if it could not communicate?

**Comment by aschmois** [comment]
I'm leaving that up to the thread. It might hang if it can't communicate but threading it would be really complicated. I'm not worrying too much because this is done only once (maybe twice) and with a decent internet connection I experienced absolutely no hangs.

File: src/main/java/com/forgeessentials/core/ForgeEssentials.java [hunk 2]
```diff
@@ -76,7 +76,7 @@
     private File FEDIR;
 
     private ConfigManager configManager;
-
```
Comment (luacs1998) [inline src/main/java/com/forgeessentials/core/ForgeEssentials.java]: No

**Comment by aschmois** [comment]
Closing and squashing all commits.

**Comment by aschmois** [comment]
Ok so I squashed all commits since, it looks like this should be the final version.

**Comment by liach** [comment]
Why not merge?

**Comment by olee** [comment]
@aschmois can you rebase the PR onto the latest changes so we can merge it?
The current PR is not automatically mergable any more....

**Comment by aschmois** [comment]
@olee Sure thing.

@liach There's currently a bug in the client handshake and @luacs1998 should be looking into it when he gets back.

**Comment by luacs1998** [comment]
I'll just merge this first, then let it fall back to the link generation.

**Pull request merged**